test/test.googleauth.ts

```diff
@@ -1250,6 +1250,24 @@ describe('._checkIsGCE', () => {
     assert.equal(false, auth.isGCE);
   });
 
+  it('should retry the check for isGCE if it fails the first time',
+     async () => {
+       const auth = new GoogleAuth();
+       assert.notEqual(true, auth.isGCE);
+       // the first request will fail
+       const scope1 =
+           nock('http://metadata.google.internal').get('/').reply(500);
+       // the second one will succeed
```

Comment:
Add another check that 404s.

Reply:
Added a test for the 404, and you know, the code that actually prevents retry on a 404 :)